core/src/main/java/org/apache/metamodel/CompositeDataContext.java

```diff
@@ -190,9 +183,7 @@ public Schema getSchemaByNameInternal(String name) throws MetaModelException {
                 }
             }
         }
-        String[] result = set.toArray(new String[set.size()]);
-        Arrays.sort(result);
-        return result;
+        return new ArrayList<>(set);
```

Comment:
This is a Treeset now, see line 175, order is insertion order.

The real Question here is: should getSchemas() be List or Collection?